fabtests/functional/unexpected_msg.c

```diff
@@ -71,9 +79,25 @@ static int alloc_bufs(void)
 	rx_buf = buf;
 	tx_buf = (char *) buf + rx_size * concurrent_msgs;
 
-	if (fi->domain_attr->mr_mode & FI_MR_LOCAL) {
-		ret = fi_mr_reg(domain, buf, buf_size, FI_SEND | FI_RECV,
-				 0, FT_MR_KEY, 0, &mr, NULL);
+	no_mr_reg = (((!(fi->domain_attr->mr_mode & FI_MR_LOCAL) &&
+		       !(opts.options & FT_OPT_USE_DEVICE)) ||
+		      (!(fi->domain_attr->mr_mode & FI_MR_HMEM) &&
+		       opts.options & FT_OPT_USE_DEVICE)) &&
+		     !(fi->caps & (FI_RMA | FI_ATOMIC)));
+
+	if (!no_mr_reg) {
+		iov.iov_base = buf;
+		iov.iov_len = buf_size;
+
+		access = ft_info_to_mr_access(fi);
+
```

Comment:
nit: remove 2 blank lines above and remove 'access' variable.  Just place ft_info_to_mr_access()  directly below.  (The changes in this section has a few too many blank lines.)